frontend/src/components/Dashboard.vue

```diff
@@ -0,0 +1,321 @@
+const userRoleAssignmentApi = apiServiceFactory.getUserRoleAssignmentApi();
+const loading = ref<boolean>(false);
+const filters = ref({
+    global: { value: null, matchMode: FilterMatchMode.CONTAINS },
+    'user.user_name': { value: null, matchMode: FilterMatchMode.STARTS_WITH },
+    'role.parent_role.role_name': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+    'role.role_name': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+    'role.client_number.forest_client_number': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+});
+const userRoleAssignments = ref<FamApplicationUserRoleAssignmentGet[]>();
+
+const confirmRemoveMessage = reactive({
+    userName: '',
+    role: '',
+});
+
+onMounted(async () => {
+    // Reload list each time we navigate to this page to avoid forcing user to refresh if their access changes.
+    try {
+        loading.value = true;
+        applicationsUserAdministers.value = (
+            await applicationsApi.getApplications()
+        ).data;
+    } catch (error: any) {
+        return Promise.reject(error);
+    } finally {
+        loading.value = false;
+    }
+
+    if (isApplicationSelected) {
+        getAccessList();
+    }
+});
+const filteredOptions = computed(() => {
+    return applicationsUserAdministers.value;
+});
+
+const selectApplication = (e: DropdownChangeEvent) => {
+    setSelectedApplication(e.value ? JSON.stringify(e.value) : null);
+    if (isApplicationSelected) {
+        getAccessList();
+    }
+};
+
+async function getAccessList() {
+    try {
+        loading.value = true;
+        const list = (
+            await applicationsApi.getFamApplicationUserRoleAssignment(
+                selectedApplication.value!.application_id
+            )
+        ).data;
+        userRoleAssignments.value = list.toSorted((first, second) => {
+            const nameCompare = first.user.user_name.localeCompare(
+                second.user.user_name
+            );
+            if (nameCompare != 0) return nameCompare;
+            const roleCompare = first.role.role_name.localeCompare(
+                second.role.role_name
+            );
+            return roleCompare;
+        });
+    } catch (error: unknown) {
+        router.push('/dashboard');
+        return Promise.reject(error);
+    } finally {
+        loading.value = false;
+    }
+}
+
+async function tryDelete(assignment: FamApplicationUserRoleAssignmentGet) {
+    confirmRemoveMessage.role = assignment.role.role_name;
+    confirmRemoveMessage.userName = assignment.user.user_name;
+    useNotificationMessage.isNotificationVisible = false;
+    useNotificationMessage.notificationMsg = '';
+    confirm.require({
+        group: 'templating',
+        icon: 'none',
+        header: 'Remove Access',
+        rejectLabel: 'Remove',
+        acceptLabel: 'Cancel',
+        reject: () => {
```

Comment:
I thought this Confirm dialog is to "accept" the confirmation to remove access.
Why is the label for user to "accept" his action is labeling "Cancel" (at the "acceptLabel")?
Same as the callback function "reject: () =>" when user reject his action to remove access, the callback actually trying to remove the access. Isn't that contradict what user wants?
And that will make other developer to wrap the head around why the function/label are doing the opposite?

Reply:
According to @J0taFerreira this is because the PrimeVue component for the dialog has the button layout inverted than what we need. One idea could be to create a middle component that inverts it, making it more clear to who is using it. What do you think?